Assets/Plugins/Source/Editor/Windows/LogLevelWindow.cs

```diff
@@ -0,0 +1,114 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+using UnityEngine;
+using UnityEditor;
+using System;
+using System.Collections.Generic;
+
+#if !EOS_DISABLE
+namespace PlayEveryWare.EpicOnlineServices.Editor.Windows
+{
+    public class LogLevelWindow : EOSEditorWindow
+    {
+        LogLevelConfig currentLogLevelConfig;
+
+        string[] categories = LogLevelUtility.LogCategoryStringArray;
+        string[] levels = LogLevelUtility.LogLevelStringArray;
+
+        int selectedCategoryIndex = 0;
+        bool showAllCategories = false;
+        string previousAllCategoryLogLevel;
+
+        [MenuItem("Tools/EOS Plugin/Log Level")]
+        public static void ShowWindow()
+        {
+            GetWindow<LogLevelWindow>("Log Level");
+        }
+        protected override void Setup()
+        {
+            currentLogLevelConfig = EpicOnlineServices.Config.Get<LogLevelConfig>();
+
+            // Config.Get<T> only guarantees T to be not null, but not for its members
+            if (currentLogLevelConfig.logCategoryLevelPairs == null) 
+            {
+                currentLogLevelConfig.logCategoryLevelPairs = new List<LogCategoryLevelPair>();
+            }
+
+            // Initialize the list if config categories does not match the SDK categories, which might happen when using a different SDK version
+            if (currentLogLevelConfig.logCategoryLevelPairs.Count != categories.Length)
+            {
+                currentLogLevelConfig.logCategoryLevelPairs.Clear();
+
+                foreach (var category in categories)
+                {
+                    currentLogLevelConfig.logCategoryLevelPairs.Add(new LogCategoryLevelPair(category, "Info"));
+                }
+
+                currentLogLevelConfig.Write(true);
+            }
+        }
+        protected override void RenderWindow()
+        {
+            // Last element in the list is All Categories
+            previousAllCategoryLogLevel = currentLogLevelConfig.logCategoryLevelPairs[categories.Length - 1].level;
```

Comment:
I think this line should be moved to be between line 97 and line 98. More importantly, it should be a local variable - there is no need to store it as a member of the `LogLevelWindow` class, as it is only ever used in this circumstance - and its value does not need to persist between calls of `RenderWindow`.

Reply:
You are right, it should be local.
But it cannot be moved, because the level change should be between previous and current to check if it is dirty.